ts/components/LabelledItem.tsx

```diff
@@ -117,12 +122,24 @@ export const LabelledItem: React.FC<Props> = (props: Props) => {
     setHasFocus(false);
   };
 
-  const descriptionColor = props.isValid === false ? "red" : "bluegreyDark";
+  const onSetInputBorderColor = () => {
+    if (props.type === "text" && props.inputProps.disabled) {
+      return IOColors.greyLight;
+    }
+    if (hasFocus && isEmpty) {
+      return variables.itemBorderDefaultColor;
+    }
+
+    return props.focusBorderColor;
+  };
+
+  const descriptionColor =
+    props.type === "text" && props.inputProps.disabled
+      ? "bluegreyLight"
+      : props.isValid === false
+      ? "red"
+      : "bluegreyDark";
```

Comment:
_disclaimer_: I'm not a fan of nested ternary operators because I find them difficult to read 😬 so bear with me 🙏 

The condition `props.type === "text" && props.inputProps.disabled` is repeated 5 times in this code. I suggest you evaluate it once to allow for easier updates in the future.
It could be something along the lines of `const isDisabledTextInput = props.type === "text" && props.inputProps.disabled` to help the reader to understand the intent.